Rewrite data lakehouse guide, remove references to lakehouse (#5778)
ClickHouse/clickhouse-docs: docs/cloud/onboard/01_discover/02_use_cases/03_data_warehousing.md

---
slug: /cloud/get-started/cloud/use-cases/data_lake_and_warehouse
title: 'Data warehousing'

> description: 'Build modern data warehousing architectures with ClickHouse Cloud combining the flexibility of data lakes with database performance'

Reviewer: ```suggestion
description: 'Build modern data warehousing architectures with ClickHouse Cloud combining the flexibility of data lakes with data warehouse performance'
```

import Image from '@theme/IdealImage';
import datalakehouse_01 from '@site/static/images/cloud/onboard/discover/use_cases/datalakehouse_01.png';

Data lakes provide flexible, scalable storage for all types of data on cloud
object storage. By adding open table formats and a high-performance query engine
like ClickHouse, you can bring database-grade capabilities — ACID transactions,
schema enforcement, and fast analytical queries to your data lake without
sacrificing its flexibility.

This combination supports traditional analytics and modern AI/ML workloads in
a unified platform, giving you warehouse-level performance on top of open,

> cost-effective lake storage.

Reviewer: ```suggestion
This combination brings performance together with interoperable, cost effective storage to support your traditional analytics and modern AI/ML workloads.
```

:::tip ClickHouse Academy
Take the free [Data Warehousing with ClickHouse](https://clickhouse.com/learn/data-warehousing) course to learn more.
:::

## Components of a data lake architecture {#components-of-the-architecture}

A modern data lake architecture has several distinct but interconnected
layers that combine the flexibility of lake storage with data warehouse
capabilities. You can substitute components and evolve each layer
independently.

> <Image img={datalakehouse_01} alt="Components of a data lake architecture" size="md"/>

Reviewer 2: Let's replace this diagram with the one from here: https://clickhouse.com/use-cases/data-warehousing

| **Table format layer**  | The [table format layer](https://clickhouse.com/engineering-resources/open-table-formats) manages the logical organization of data files into tables, providing database-like features such as ACID transactions, schema enforcement and evolution, time travel, and performance optimizations like data skipping and clustering. |
| **Object storage**      | This layer provides scalable, durable, and cost-effective storage for all data files and metadata. It handles the physical persistence of data in an open format, enabling direct access from multiple tools and systems.                                                                                                                          |
| **Client applications** | Various tools and applications that connect to the platform to query data, visualize insights, or build data products. These can include BI tools, data science notebooks, custom applications, and ETL/ELT tools.                                                                                                                            |

> ## Benefits of combining data lakes with a query engine {#benefits-of-this-architecture}

Reviewer: ```suggestion
## Benefits of combining data lakes with a data warehouse {#benefits-of-this-architecture}
```
